### ISSUE TITLE
Decoder training

### ISSUE THREAD

**Issue by kilotaras** [yingcong/Facelet_Bank#6]
Hi. I'm trying to reproduce your results on smaller images, but it seems that decoder doesn't work correctly with such sizes

Can you expand on how you trained `vgg_decoder`?

**Comment by yingcong** [comment]
Let **vgg_feat1, vgg_feat2, vgg_feat3 = VGG(image_input)** be the VGG network defined in 
https://github.com/yingcong/Facelet_Bank/blob/0f2edb49ea72b4da5da6a12d08cbcd4ff7811f8f/network/base_network.py#L84
where vgg_feat1, vgg_feat2, vgg_feat3 are the vgg features of ReLU3_1, ReLU4_1 and ReLU5_1

Let **image_dec = Decoder(feat1, feat2, feat3)** be the decoder defined in 
https://github.com/yingcong/Facelet_Bank/blob/0f2edb49ea72b4da5da6a12d08cbcd4ff7811f8f/network/decoder.py#L13

The released decoder model is trained with the following procedure:

1. Extract several kinds of attribute vectors (Younger, Older, Facehair and  Gender is extracted in our implementation) with "DFI/demo2_facelet.py". For each sample, it will extract 3 attribute vectors (denoted as v1, v2, v3) corresponding to ReLU3_1, ReLU4_1 and ReLU5_1 layers. 

2. We train the network as follows:

- given a training sample **image_input**, we first get
**vgg_feat1, vgg_feat2, vgg_feat3 = VGG(image_input)**

- produce a random number **s**

- Compute
**image_dec=Decoder(vgg_feat1+s\*v1, vgg_feat2+s\*v2, vgg_feat3+s\*v3)** 
and 
**feat1_out, feat2_out, feat3_out = VGG(image_dec)**

- Minimize the loss function:
**L=||vgg_feat1+s\*v1 - feat1_out||^2 +||vgg_feat2+s\*v2 - feat2_out||^2 + ||vgg_feat3+s\*v3 - feat3_out||^2**

**Comment by fjzpcmj** [comment]
Hi，i read your paper, the VGG encoder is pretrained in ImageNet dataset.  Then the VGG encoder is fixed? Did you finetune the VGG encoder in some face datesets? And is the encoder fixed when the decoder is trained?

**Comment by yingcong** [comment]
Yes, it is fixed. I have tried to fine-tune VGG with simple identity classification or attribute classification tasks, but gets inferior results.
